ewkcoffea/modules/corrections.py

```diff
@@ -144,65 +148,120 @@ def run3_muons_sf_Attach(muons,year,syst,id_method,iso_method):
     abseta_flat = ak.flatten(abs(muons.eta))
     pt_flat = ak.flatten(muons.pt)
 
-    # For now, cap all pt at 199.9 (limit for this particular sf)
+    # For now, cap all pt at 199.9 and min of 15.0 (limits for this particular json)
     pt_flat = ak.where(pt_flat>199.9,199.9,pt_flat)
     pt_flat = ak.where(pt_flat<15.0,15.0,pt_flat)
 
     # Evaluate the SF
     ceval = correctionlib.CorrectionSet.from_file(fname)
-    sf_id_flat = ceval[id_method].evaluate(abseta_flat,pt_flat,syst)
-    sf_iso_flat = ceval[iso_method].evaluate(abseta_flat,pt_flat,syst)
-    sf_flat = sf_id_flat * sf_iso_flat
-    sf = ak.unflatten(sf_flat,ak.num(muons.pt))
+    sf_id_flat_nom = ceval[id_method].evaluate(abseta_flat,pt_flat,"nominal")
+    sf_id_flat_syst = ceval[id_method].evaluate(abseta_flat,pt_flat,"syst")
+    sf_id_flat_hi = sf_id_flat_nom + sf_id_flat_syst
+    sf_id_flat_lo = sf_id_flat_nom - sf_id_flat_syst
+
+    sf_iso_flat_nom = ceval[iso_method].evaluate(abseta_flat,pt_flat,"nominal")
+    sf_iso_flat_syst = ceval[iso_method].evaluate(abseta_flat,pt_flat,"syst")
+    sf_iso_flat_hi = sf_iso_flat_nom + sf_iso_flat_syst
+    sf_iso_flat_lo = sf_iso_flat_nom - sf_iso_flat_syst
+
+    sf_flat_nom = sf_id_flat_nom * sf_iso_flat_nom
+    sf_flat_hi = sf_id_flat_hi * sf_iso_flat_hi
+    sf_flat_lo = sf_id_flat_lo * sf_iso_flat_lo
+
+    sf_nom = ak.unflatten(sf_flat_nom,ak.num(muons.pt))
+    sf_hi = ak.unflatten(sf_flat_hi,ak.num(muons.pt))
+    sf_lo = ak.unflatten(sf_flat_lo,ak.num(muons.pt))
 
-    muons['ele_sf'] = ak.ones_like(sf)
-    muons['muon_sf'] = sf
+    muons['sf_nom_muon'] = sf_nom
+    muons['sf_hi_muon']  = sf_hi
+    muons['sf_lo_muon']  = sf_lo
+    muons['sf_nom_elec'] = ak.ones_like(sf_nom)
+    muons['sf_hi_elec']  = ak.ones_like(sf_nom)
+    muons['sf_lo_elec']  = ak.ones_like(sf_nom)
 
-def run3_electrons_sf_Attach(electrons,year,valtype,wp):
+
+def run3_electrons_sf_Attach(electrons,year,wp):
```

Comment:
Same comment as above (lower case function names, to try to be consistent with python style guide).